.github/workflows/optimize-imgs.yml

```diff
@@ -0,0 +1,35 @@
+# https://github.com/calibreapp/image-actions
+name: Compress Images
+on:
+  pull_request:
+    # Run Image Actions when JPG, JPEG, PNG or WebP files are added or changed.
+    # See:
+    #   - https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#onpushpull_requestpaths.
+    paths:
+      - '**.jpg'
+      - '**.jpeg'
+      - '**.png'
+      - '**.webp'
+jobs:
+  build:
+    # Uncomment below line to  run on Pull Requests within the same repository,
+    # and not from forks.
+    # if: github.event.pull_request.head.repo.full_name == github.repository
+    name: calibreapp/image-actions
+    permissions: write-all
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v4
+
+      - name: Compress Images
+        uses: calibreapp/image-actions@main
+        with:
+          # The `GITHUB_TOKEN` is automatically generated by GitHub and scoped
+          # only to the repository that is currently running the action. By
+          # default, the action can’t update Pull Requests initiated from
+          # forked repositories.
+          # See:
+          #   - https://docs.github.com/en/actions/reference/authentication-in-a-workflow
+          #   - https://help.github.com/en/articles/virtual-environments-for-github-actions#token-permissions
+          githubToken: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
This is kind of too much assumptions happening, but if we run this action on commit to main only, then this would not be needed. Nor would it be needed if we only do this for PRs created by people with access to this repo. Only for external fork PR this is needed. So if we keep this for PR, even then we can get rid of this line. 

I prefer running it on main tho, so if I do any commit bypassing PR workflow even then it runs. And this was it will run fewer times over all tho, as we may have many commits in PR and only one push to main per PR.

Reply:
See https://github.com/fastn-stack/fastn/pull/2016/commits/e0f09c84156d522c2ffb87cdd1b0be475f62c90b. I've set this to run on push to `main` now. A separate PR with optimized images will be created if there's any output.